src/test/scala/sigmastate/utxo/ErgoLikeTestInterpreterSpecification.scala

```diff
@@ -614,4 +615,27 @@ class ErgoLikeInterpreterSpecification extends SigmaTestingCommons {
 
     //todo: check failing branches
   }
+
+  property("DeserializeRegister value type mismatch") {
+    val prover = new ErgoLikeTestProvingInterpreter
+
+    val sigmaProp = SigmaPropConstant(prover.dlogSecrets.head.publicImage)
+    // put SigmaProp into the register
+    val regValue = ByteArrayConstant(ValueSerializer.serialize(sigmaProp))
+    val box = ErgoBox(20, TrueLeaf, 0, Seq(), Map(R4 -> regValue))
+
+    // expect SBoolean in the register
+    val prop = DeserializeRegister(R4, SBoolean)
+
+    val ctx = ErgoLikeContext(
+      currentHeight = 50,
+      lastBlockUtxoRoot = AvlTreeData.dummy,
+      minerPubkey = ErgoLikeContext.dummyPubkey,
+      boxesToSpend = IndexedSeq(box),
+      ErgoLikeTransaction(IndexedSeq(), IndexedSeq(ErgoBox(10, prover.dlogSecrets(2).publicImage, 0))),
```

Comment:
@greenhat What is this output box means?

Reply:
@aslesarenko Remnants of a copy-paste from a test. Thanks. Fixed.